crates/telio-wg/src/wg.rs

```diff
@@ -388,9 +389,12 @@ impl WireGuard for DynamicWg {
         let _ = self.task.stop().await.resume_unwind();
     }
 
-    async fn reset_existing_connections(&self, exit: PublicKey) -> Result<(), Error> {
+    async fn reset_existing_connections(&self, exit: &ExitNode) -> Result<(), Error> {
+        let pubkey = exit.public_key;
+        let addr = exit.endpoint.map(|addr| addr.ip());
```

Comment:
There is a small problem with using exit_node's endpoing like this. So generally `ExitNode` struct, which is provided to `connect_exit_node(...)` structure may potentially refer to VPN server as well as Meshnet exit node. While for VPN apps will know, and will provide endpoint to connect to, this is not the case for meshnet exit nodes.

What is slightly annoying for meshnet nodes - the endpoint is fundamentally rather dynamic, and can change during runtime at any point. Therefore my recommendation would be to resolve the endpoint in the `device/mod.rs` method `connect_exit_node()` just before calling `.reset_existing_connections()`. The resolution would look like this. If `exit_node` public key is amongst the `self.requested_state_meshnet_config.peers` or even better -> in wireguard peer list, which you can acquire from `self.entities.wireguard_interface`, then acquire endpoint from that peers endpoint provided by `self.entities.wireguard_interface`. Otherwise take endpoint from `exit_node`.

Reply:
Why it's not the case for the meshnet node?

Reply:
@Jauler I've proved fallback in case it's `None`. However, your description provides a more complete solution :+1: